usr.bin/truss/syscalls.c

```diff
@@ -1514,7 +1521,7 @@ print_cmsgs(FILE *fp, pid_t pid, bool receive, struct msghdr *msghdr)
 	}
 	cmsgbuf = calloc(1, len);
 	if (get_struct(pid, (uintptr_t)msghdr->msg_control, cmsgbuf, len) == -1) {
-		fprintf(fp, "%p", msghdr->msg_control);
+		print_pointer(fp, (uintptr_t)msghdr->msg_control);
```

Comment:
@brooksdavis does `PRIxPTR` do the right thing for uintptr_t in purecap or does it print the offset?

Reply:
You changed %P to always print the address in 1322275559ae457e08b299290f4b3744e7394343.  Looking at plain '%p', it seems to have always printed the address and not the offset.